src/TextField.jsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import cx from 'classnames';
+
+/**
+ * @module TextField
+ */
+class TextField extends React.Component {
+
+	constructor(props) {
+		super(props);
+		this.state = {
+			value: props.value || '',
```

Comment:
I think this is how value should be set so you don't have to specify it as a prop if `undefined` and so you won't get the controlled/uncontrolled component warning.